sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1307,18 +1309,42 @@ fn check_param_type<Mode: ExecutionMode>(
                 Mode::allow_arbitrary_values() || !abilities.has_key(),
                 "Raw value should never be an object"
             );
-            ty
+                if ty != param_ty {
+                    return Err(command_argument_error(
+                        CommandArgumentError::TypeMismatch,
+                        idx,
+                    ))
+                }
+        }
+        Value::Object(obj) =>  {
+            let ty = &obj.type_;
+            if ty != param_ty {
+                return Err(command_argument_error(
+                    CommandArgumentError::TypeMismatch,
+                    idx,
+                ))
+            }
+        }
+        Value::Receiving(_, _) =>  {
+            if let Type::StructInstantiation(sidx, targs) = param_ty {
```

Comment:
nit, maybe flip this to a `let-else` I found it a bit confusing that the valid case was inside the if and the error case was the default/fallthrough